src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -151,6 +160,16 @@ class BlockchainImpl(
 
   override def getTotalDifficultyByHash(blockhash: ByteString): Option[BigInt] = totalDifficultyStorage.get(blockhash)
 
+  override def getAccount(address: Address, blockNumber: BigInt): Option[Account] =
+    getBlockHeaderByNumber(blockNumber).flatMap { bh =>
+      val mpt = new MerklePatriciaTrie[Address, Account](
+        Some(bh.stateRoot.toArray),
+        nodeStorage,
+        crypto.kec256(_: Array[Byte])
+      )
+      mpt.get(address)
+    }
```

Comment:
@ntallar the correct (hopefully) impl of `getAccount`. I'm still not sure if `Blockchain` is the best place to put this method.